pinot-broker/src/main/java/org/apache/pinot/broker/routing/segmentpruner/PartitionSegmentPruner.java

```diff
@@ -148,12 +149,12 @@ public synchronized void refreshSegment(String segment) {
   }
 
   @Override
-  public List<String> prune(BrokerRequest brokerRequest, List<String> segments) {
+  public Set<String> prune(BrokerRequest brokerRequest, Set<String> segments) {
```

Comment:
Let's see if the pruner code can deal with set v/s list as opposed to changing the APIs across all pruner users.